### ISSUE TITLE
Introduce preserveDrawingBuffer attribute for webgl context

### ISSUE THREAD

**Issue by mattrei** [tangrams/tangram#526]
Hello everyone,

!Heureka! 
I have found the solution to all of my todays Tangram problems ;).

My use case is to put the contents of the HTML Tangram _canvas_ as a texture on a THREE.Js Mesh.

But for this to work, I need to set the _preserveDrawingBuffer_ attribute to _true_ when obtaining a _webgl_ context. 
Tangram does this in the _scene.js_ on line 226. However I cannot pass the _perserveDrawingBuffer_ option. 

May you add this feature please? So that it is configurable as an API? I could do a PR but you are the experts.

However for me this a major show stopper in using Tangram JS!

Thx for your expertise on this.

**Comment by bcamper** [comment]
Hi -- yes we can look at providing an option for this. As you know it will affect rendering performance so it would remain off by default, but it's reasonable to let the user enable this if desired.

**Comment by bcamper** [comment]
Added in ccea85d50e7569020afd0263b987d3cf2bd132bb, will be released in v0.12.1.

WebGL context options can be passed when creating the Tangram Leaflet layer:

```
layer = Tangram.leafletLayer({
  scene: ...,
  webGLContextOptions: { // explicitly add/override WebGL context options
    preserveDrawingBuffer: true
  }
});
```

**Comment by bcamper** [comment]
Released in [v0.12.1](https://github.com/tangrams/tangram/releases/tag/v0.12.1)
